primitives/src/math/fixed.rs

```diff
@@ -243,36 +243,35 @@ impl<F: Fixed + ToString, N: TryFrom<u128>> FromFixedToDecimal<F> for N {
     fn from_fixed_to_fixed_decimal(fixed: F, decimals: u8) -> Result<N, &'static str> {
         let decimals_usize = decimals as usize;
         let s = fixed.to_string();
-        let mut parts: Vec<&str> = s.split('.').collect();
-        // If there's no fractional part, then `fixed` was an integer.
-        if parts.len() != 2 {
-            parts.push("0");
+        let mut parts = s.split('.');
+        let int_part = parts.next().ok_or("Empty string provided")?;
+        let frac_part = parts.next().unwrap_or("0");
+        // Ensure that the iterator is now exhausted.
+        if parts.next().is_some() {
+            return Err("The string contains multiple decimal points");
         }
 
-        let (int_part, frac_part) = (parts[0], parts[1]);
-        let mut increment = false;
-
+        let mut increment_int_part = false;
         let new_frac_part = if frac_part.len() < decimals_usize {
             format!("{}{}", frac_part, "0".repeat(decimals_usize.saturating_sub(frac_part.len())))
         } else {
```

Comment:
```suggestion
        } else if frac_part.len() > decimal_usize {
```

> Note that `round_digit` might be `None`, for example if `frac_part = 245` and `decimals = 3`. There's no need to round in that case.

So I assume it's always the case that if `frac_part.len() == decimals` then we don't need to round.